test/framework/resources/k8s/resources/multus_resources.go

```diff
@@ -0,0 +1,320 @@
+package resources
+
+import (
+	"context"
+
+	"github.com/aws/amazon-vpc-cni-k8s/test/framework/resources/k8s/manifest"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	v1 "k8s.io/api/rbac/v1"
+
+	apiextensionsV1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
+	apiextensionsclient "k8s.io/apiextensions-apiserver/pkg/client/clientset/clientset"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/rest"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type MultusResourceManager interface {
+	CreateClusterRole() error
```

Comment:
I feel we should move all these functions into `test/framework/resources/k8s/resources/common` or `utils` maybe(?) and just pass the resource manager information so that say functions like `DeleteDaemonSet` can be common and we can use it for any type of DS. Functions which are very specific to a DS can be placed here. Thoughts on this?

Reply:
Yes, I had thought about this. The multus ds turned out to have lot fo fields and the existing ds had too few fields in this comparison. Will check how I can make it more extensible.

Reply:
Thank you :)